goal_src/jak1/pc/features/speedruns.gc

```diff
@@ -387,6 +388,21 @@
 (defun speedrun-mode-update ()
   "A per frame update for speedrunning related stuff"
   (when (-> *pc-settings* speedrunner-mode?)
+    ;; Check if run/autosplitter should be started (out of blackout and player has control of jak)
+    (when (and (not (-> *speedrun-info* has-started?))
+               *target*
+               ;; not in blackout
+               (>= (-> *display* base-frame-counter) (-> *game-info* blackout-time))
+               (!= (-> *setting-control* current bg-a-force) 1.0)
+               ;; target has landed on ground/water, or we're about to enter target-hit-ground state
+               (or (logtest? (-> *target* control status) (cshape-moving-flags onground onsurf on-water))
+                   (= (-> *target* next-state name) 'target-hit-ground)))
+      (true! (-> *speedrun-info* has-started?))
+      ;; spawn a process that resets autosplitter after suspending for 1 frame
+      (process-spawn-function process
+        (lambda ()
+          (suspend)
+          (update-autosplit-jak1-new-game))))
```

Comment:
This 1 frame suspend is a best-attempt to make the autosplit reset line up with the visual frame when the run should start. Sometimes livesplit's memory scanning ends up being a frame or two delayed picking this up, but the 1 frame suspend at least ensures it doesn't reset 1 frame early.